src/AutoRest.CSharp.V3/Output/Models/Types/ObjectType.cs

```diff
@@ -333,6 +333,12 @@ private IEnumerable<ObjectTypeProperty> BuildProperties()
             foreach (Property property in _objectSchema.Properties!)
             {
                 var name = property.CSharpName();
+                // Resolve property-type name conflicts
+                // TODO: we might eventually need a more generic version of this logic
+                if (name == Type.Name)
+                {
+                    name += "Value";
```

Comment:
This works for 99% of the cases, but keep in mind that if, somehow, there was a `[TypeName]` property and a `[TypeName]Value` property already existing, this would cause a different kind of conflict. I've seen weird stuff like that situation from Network before. So, don't be surprised if it happens.